uavf_2024/imaging/target_tracker.py

```diff
@@ -33,8 +33,31 @@ def closest_match(self, target_desc: TargetDescription) -> Target3D:
         '''
         best_match, best_score = self._targets[0], 0
 
+        
+        '''Converts the target description value from an int into a numpy prob distribution array to be dot producted with the list of targets 
+            '''
+        target_desc_list = list(vars(target_desc).values())
+
+        target_desc = TargetDescription(
+                    np.eye(13)[target_desc_list[0]],
+                    np.eye(35)[target_desc_list[1]],
+                    np.eye(8)[target_desc_list[2]],
+                    np.eye(8)[target_desc_list[3]],
+                )
+        
+        for target_detail, target_probability in vars(target_desc).items():
+                target_probability[ target_probability == 0 ] = 10 ** -4 
+        
         for target in self._targets:
+            # converts all 0's in the prob numpy array for target into a small number and then back after the comparison
+            for target_detail, target_probability in vars(target.description).items():
+                target_probability[ target_probability == 0 ] = 10 ** -4 
+
             new_score = calc_match_score(target_desc, target.description)
```

Comment:
Now that I've thought about it some more, can you move the logic that adds a small amount to each distribution into `calc_match_score` instead of where it is right now? This way we can reuse it wherever we use that comparison function instead of needing to write it again every time.